src/main/java/org/reaktivity/nukleus/http/internal/stream/ConnectionPool.java

```diff
@@ -31,25 +31,27 @@
 {
     public enum CloseAction
     {
-        END, ABORT;
-    };
+        END, ABORT
+    }
     private final Deque<Connection> availableConnections;
     private final String connectName;
     private final long connectRef;
     private final ClientStreamFactory factory;
 
     private int connectionsInUse;
-    private ConnectionRequest nextRequest;
 
     ConnectionPool(ClientStreamFactory factory, String connectName, long connectRef)
     {
         this.factory = factory;
         this.connectName = connectName;
         this.connectRef = connectRef;
-        this.availableConnections = new ArrayDeque<Connection>(factory.maximumConnectionsPerRoute);
+        this.availableConnections = new ArrayDeque<>(factory.maximumConnectionsPerRoute);
     }
 
-    public void acquire(ConnectionRequest request)
+    /*
+     * @return true if a connection is acquired, otherwise false
+     */
+    boolean acquire(ConnectionRequest request)
```

Comment:
Would it make sense to return the `Connection` here, and then test for `null` outside instead of the boolean return type?